src/TransformationUtils.cc

```diff
@@ -88,57 +88,60 @@ std::unique_ptr<TransitionSystem> toTransitionSystem(ChcDirectedGraph const & gr
     return ts;
 }
 
-bool strongConnection(std::unordered_set<int> & visitedVertices, std::unordered_set<int> & verticesOnStack,
-                      AdjacencyListsGraphRepresentation const & graphRepresentation, ChcDirectedGraph const & graph,
-                      SymRef node) {
-    visitedVertices.insert(node.x);
-    verticesOnStack.insert(node.x);
+// This function follows a Tarjan's strong connection detection algorithm:
+// https://en.wikipedia.org/wiki/Tarjan's_strongly_connected_components_algorithm
+// If some vertice is added on stack and it was already visited, then the strong connection is detected
+// It was updated to remember the nodes which participate in the loop (lines 113, 119)
```

Comment:
Are these line numbers correct?
Line 119 is `break`.
We probably should not refer to line numbers in here, it can very easily become incorrect.
Also, please replace `strong connection` with `strongly connected component`.

Reply:
Done